### PR TITLE
HWP-222: Changed redux code to 'Duck' structure

### PR DESCRIPTION
**Pull request by ZachBourque**
Applied the 'Duck' redux structure, which refactors redux code into singular files specific to the type of actions they are performing. (postDuck, communityDuck, etc.)

Fixed an issue in which the useEffect hook would run constantly through the lifecycle of the function.

This PR includes the following proposed change(s):

- HWP-222

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. << @ZachBourque 

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/PostsList.js [hunk 1]
```diff
@@ -45,6 +47,7 @@ const PostsList = (props) => {
               }}
               variant="outlined"
               square>
+              <h1>Testing2</h1>
```
Comment (BradyMitch) [inline app/client/src/components/PostsList.js:50]:
```suggestion
```

**Comment by BradyMitch** [comment]
Remove changes made to app/server/express.js

**Comment by CITZ-rwk** [comment]
approved

**Review by BradyMitch — APPROVED** [review]
Tested, great job

**Pull request merged**